### PR TITLE
Add default init container ResourceRequirements

### PR DESCRIPTION
**Pull request by czheng94**
This change is [Reviewable](https://reviewable.io/reviews/kubeflow/mpi-operator/142)

### CODE REVIEW

File: pkg/controllers/v1alpha2/mpi_job_controller.go [hunk 1]
```diff
@@ -1198,6 +1201,16 @@ func (c *MPIJobController) newLauncher(mpiJob *kubeflow.MPIJob, kubectlDeliveryI
 				MountPath: configMountPath,
 			},
 		},
+		Resources: corev1.ResourceRequirements{
```
Comment (terrytangyuan) [inline pkg/controllers/v1alpha2/mpi_job_controller.go]: Should we expose this to `MPIJobSpec` similar to `MPIReplicaSpecs`? What if user needs resources other than cpu and memory?

Comment (yuzisun) [reply pkg/controllers/v1alpha2/mpi_job_controller.go]: This is only for init container though, not sure if user needs to customize it.

Comment (terrytangyuan) [reply pkg/controllers/v1alpha2/mpi_job_controller.go]: What if their requirements are different, e.g. larger than these default values?

Comment (czheng94) [reply pkg/controllers/v1alpha2/mpi_job_controller.go]: In what cases would user want to customize the init container?

Comment (terrytangyuan) [reply pkg/controllers/v1alpha2/mpi_job_controller.go]: @gaocegege @rongou Any thoughts on this?

Comment (rongou) [reply pkg/controllers/v1alpha2/mpi_job_controller.go]: Seems fine. The init container really just copies the `kubectl` binary and does some initial checking. Resource usage should be minimal.

**Pull request merged**